### PR TITLE
UUID package improvements

### PR DESCRIPTION
**Pull request by stevvooe**
To avoid errant log messages, we've disabled logging by default when uuid is having trouble accessing `/dev/urandom`. This also takes the work from #786 and incorporates the feedback.

This carries #784 and #786.

cc @jfrazelle @tiborvass @ibuildthecloud

### CODE REVIEW

File: uuid/uuid.go [hunk 1]
```diff
@@ -30,7 +29,7 @@ var (
 
 	// Loggerf can be used to override the default logging destination. Such
 	// log messages in this library should be logged at warning or higher.
-	Loggerf = log.Printf
+	Loggerf func(format string, args ...interface{}) = func(format string, args ...interface{}) {}
```
Comment (tiborvass) [inline uuid/uuid.go]: no need for type, it will infer from value

**Comment by aaronlehmann** [comment]
Do we need a check on whether Loggerf is nil in Generate, before it is used?

**Comment by tiborvass** [comment]
@aaronlehmann +1

**Comment by stevvooe** [comment]
@aaronlehmann No, because it is initialized.

File: uuid/uuid.go (hunk 1 above)
Comment (stevvooe) [inline uuid/uuid.go]: fixed.

**Comment by tiborvass** [comment]
LGTM

**Comment by aaronlehmann** [comment]
Ah, missed the fact that Loggerf is initialized. LGTM.

**Comment by stevvooe** [comment]
@tiborvass Are you going to update in docker/docker?

**Pull request merged**